File: 프로그래머스/2/42578. 의상/의상.java
import java.util.ArrayList;
import java.util.HashMap;

class Solution {
    public int solution(String[][] clothes) {
        HashMap<String, ArrayList<String>> map = new HashMap<>();
        
        for (String[] cloth : clothes) {
            ArrayList<String> list = map.getOrDefault(cloth[1], new ArrayList<>());
            list.add(cloth[0]);
            map.put(cloth[1], list);
        }
        
        int total = 1;
        
        for (ArrayList<String> list : map.values()) {
            total *= list.size() + 1;
        }
            
        return total - 1;
    }
    
}

File: 프로그래머스/2/42578. 의상/의상.kt
class Solution {
    fun solution(clothes: Array<Array<String>>): Int {
        val map = HashMap<String, Int>()

        clothes.forEach {
            val (_, type) = it
            val cnt = map.getOrDefault(type, 1) + 1
            map.put(type, cnt)
        }

        return map.values.reduce { acc, i -> acc * i } - 1
    }
}
